pkg/services/live/live.go

```diff
@@ -277,6 +290,23 @@ func (g *GrafanaLive) Init() error {
 	return nil
 }
 
+func checkOrigin(r *http.Request, appURL *url.URL) bool {
```

Comment:
My only concern is when appURL = "localhost:3000"

This usually means that the appURL was not actually configured.  In the frontend, we decided to just use the browser URL rather than having an error that everyone needs to fix.  Can we do the same here?  essentially skip verification if it is not specified?

Reply:
`localhost:3000` will work since if `root_url` not configured then appURL is `http://localhost:3000/`.

Approach in this pr won't work if `root_url` set to sth like `https://mygrafana.example.com` but user tries to open `http://localhost:3000/` which is an actual upstream behind `https://mygrafana.example.com`.

In theory we can skip check if Origin starts with `http://localhost:` - in this case I can only imagine a case where a user can hijack WS connection for it's own account only. But this won't help a lot with cases where Grafana instance is accessed over ip: like `127.0.0.1:3000` or `10.0.0.1:3000` etc.